### PR TITLE
Common files improvements

### PR DESCRIPTION
**Pull request by geeknoid**
- Move many of the files into a common subdirectory, to make the repos cleaner, and to make updates to common files more robust.
 
- Generally improve linting logic.

- Retire unneeded scripts

- Introduce gobuild.sh and report_build_info.sh

- Add lint-helm support.

- lint-yaml now only applies to YAML files that don't contain {{ in order to skip templates.

### CODE REVIEW

File: files/bin/report_build_info.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  VERSION="${ISTIO_VERSION}"
+fi
+
+DOCKER_HUB="docker.io/istio"
+if [[ -n ${ISTIO_DOCKER_HUB} ]]; then
+  DOCKER_HUB="${ISTIO_DOCKER_HUB}"
+fi
+
+# used by bin/gobuild.sh
+echo "istio.io/cni/pkg/version.buildVersion=${VERSION}"
```
Comment (howardjohn) [inline files/bin/report_build_info.sh]: seems hardcoded to cni

File: files/Makefile.common.mk [hunk 2]
```diff
@@ -19,37 +19,43 @@
...
 lint-scripts:
-	@find . -path ./vendor -prune -o -type f -name '*.sh' -print0 | xargs -0 shellcheck
+	@${FINDFILES} -name '*.sh' -print0 | ${XARGS} shellcheck
 
 lint-yaml:
-	@find . -path ./vendor -prune -o -type f \( -name '*.yml' -o -name '*.yaml' \) -print0 | xargs -0 yamllint -c ./.yamllint.yml
+	@${FINDFILES} \( -name '*.yml' -o -name '*.yaml' \) -print0 | ${XARGS} grep -L -e "{{" | xargs -r yamllint -c ./.yamllint.yml
+
+lint-helm:
+	@${FINDFILES} -name 'Chart.yaml' -print0 | ${XARGS} -L 1 dirname | xargs -r helm lint --strict
```
[the remark this answers is not in the document]
Comment (geeknoid) [reply files/Makefile.common.mk]: No, it won't. Part of starting to use these various linting rules is to fix the lint issues.

File: files/bin/report_build_info.sh (hunk 1 above)
Comment (geeknoid) [reply files/bin/report_build_info.sh]: Whoops. Thanks.

**Comment by geeknoid** [comment]
@howardjohn PTAL

File: files/common/scripts/gobuild.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+SCRIPTPATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+SCRIPTDIR=$(dirname "${SCRIPTPATH}")
+
+OUT=${1:?"output path"}
+BUILDPATH=${2:?"path to build"}
+
+set -e
+
+BUILD_GOOS=${GOOS:-linux}
```
Comment (sdake) [inline files/common/scripts/gobuild.sh]: surprised this picks up the GOOS or GOARCH platforms not linux_amd64.  The GOOS nonr the GOARCH is exported from the top level makefile.

Comment (geeknoid) [reply files/common/scripts/gobuild.sh]: This file is largely copied from istio/istio/bin/gobuild.sh. I'm just starting to use it in getting istio/cni to use the build-tools container, and it's quite likely some changes will be needed before it's ready for the real-world.

**Pull request merged**